packages/e2e-test-utils-playwright/src/selectors/block-selector-engine.ts

```diff
@@ -0,0 +1,72 @@
+type BlockSelectorParams = {
+	clientId?: string;
+	name?: string;
+	tag?: string;
+	title?: string;
+};
+
+export default function createBlockSelectorEngine() {
+	// Util functions need to be defined in the `createBlockSelectorEngine`
+	// body. Playwright converts this function to a string, and won't be able
+	// to reach functions that are defined outside the body.
+	function parsePlaywrightSelector( selector: string ): BlockSelectorParams {
+		if ( ! selector ) {
+			return {};
+		}
+
+		const [ , name ] = selector.match( /^([a-z0-9\/\-]+)/ ) ?? [];
+
+		const [ , squareBracketParams ] = selector.match( /\[(.*)\]/ ) ?? [];
+		const [ , title ] = squareBracketParams?.match( /title="(.+?)"/ ) ?? [];
+		const [ , clientId ] =
+			squareBracketParams?.match( /clientId="([0-9a-f\-]+?)"/ ) ?? [];
+		const [ , tag ] = squareBracketParams?.match( /tag="(.+?)"/ ) ?? [];
+
+		return {
+			clientId,
+			name: name !== '*' ? name : undefined,
+			tag,
+			title,
```

Comment:
If we can chain this selector with others, then I don't think we need `tag` and `title` here? This leaves us only with `clientId`, I wonder if it's necessary to include that as well, since the usage of that should be really rare?

I guess what I mean here is that I prefer this selector to be as simple as possible, given the fact that it should not be recommended to use inaccessible selectors after all. We should use this sparingly when it makes sense. I feel like selecting by block types is common enough in our tests and in the context of the block editors though.